language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSPackageLoader.java

```diff
@@ -77,12 +78,12 @@ private LSPackageLoader(LanguageServerContext context) {
      *
      * @return {@link List} of local repo packages
      */
-    public List<PackageInfo> getLocalRepoPackages(PackageRepository repository) {
+    public List<ModuleInfo> getLocalRepoPackages(PackageRepository repository) {
         if (!this.localRepoPackages.isEmpty()) {
             return this.localRepoPackages;
         }
         this.localRepoPackages.addAll(checkAndResolvePackagesFromRepository(repository, Collections.emptyList(),
-                this.distRepoPackages.stream().map(PackageInfo::packageIdentifier).collect(Collectors.toSet())));
+                this.distRepoPackages.stream().map(ModuleInfo::packageIdentifier).collect(Collectors.toSet())));
```

Comment:
@malinthar why did we pass `distRepoPackages` here instead of `localRepoPackages` originally?

Reply:
If there is already a package/module with the same identifier in the distribution we don't load that from the local repo. So that's the idea of passing distRepoPackages. (Already available list of modules)

Reply:
Got it